### ISSUE TITLE
Sort/Group/Organize Connections

### ISSUE THREAD

**Issue by byjg** [mysql/mysql-shell-plugins#11]
On MySQLWorkbench, I can group my connections and organize their order by dragging and dropping the connection to desired position.

![image](https://user-images.githubusercontent.com/981924/185975611-59557fb5-760f-484f-9087-e4542ead2690.png)

Is it possible to have a similar feature on the VSCode extension?

**Comment by mike-lischke** [comment]
Yes, we like this too and it's planned for a future version.

**Comment by mbodzek** [comment]
Hello @byjg,

Here is an example how to do it properly. Suppose I have 3 connections in db:
```
sqlite> SELECT c.id, c.caption, hc.profile_id, hc.folder_path, hc.`index` 
FROM db_connection c 
LEFT JOIN profile_has_db_connection hc 
ON c.id = hc.db_connection_id;

1|Conection1|1|/|1
2|Connection2|1|/|2
3|Connection3|1|/|3
```
Now I want move Connection3 into position between 1 and 2. To do that I need update all indexes in `profile_has_db_connection` table that are grater than 1 and smaller than 3 by adding +1 to move them and make space for Connection3
```
sqlite> UPDATE profile_has_db_connection 
SET `index`=`index`+1 
WHERE profile_id=1 AND folder_path='/' AND `index`>1 AND `index`<3;
```
Next I have to update index for Connection3
```
sqlite> UPDATE profile_has_db_connection 
SET `index`=2 
WHERE profile_id=1 AND folder_path='/' AND db_connection_id=3;
```
At the end my connections are in order that I want
```
sqlite> SELECT c.id, c.caption, hc.profile_id, hc.folder_path, hc.`index` 
FROM db_connection c 
LEFT JOIN profile_has_db_connection hc 
ON c.id = hc.db_connection_id;

1|Conection1|1|/|1
2|Connection2|1|/|3
3|Connection3|1|/|2
```

For grouping connection you have to update `folder_path` in `profile_has_db_connection`.
Keep it mind that feature is not yet supported by the frontend and will not work in extension.
We have that on our todo list and eventually we'll implement it.

I hope my post will be helpful for you.

**Comment by mbodzek** [comment]
No, that's all, but seems frontend doesn't have that implemented yet.

**Comment by Hyunho-Jung** [comment]
I would like to reiterate the need for this feature as well.

**Comment by mike-lischke** [comment]
@Hyunho-Jung to signal approbation upvote the original request above, instead of adding a new comment. Thanks.
